services/QuillLMS/app/models/school.rb

```diff
@@ -72,6 +72,15 @@ class School < ApplicationRecord
            :lower_grade_greater_than_upper_grade
   validates :zipcode, length: { minimum: 5 }, allow_blank: true
 
+  # Lambda has to be wrapped in parens to avoid syntax error in this construction.
+  # Also, this is slightly magical:
+  scope :premium, (lambda do
```

Comment:
[Optional - style]

This syntax looks cleaner to me:

`  mylambda =  lambda do
    left_outer_joins(:subscriptions)
      .left_outer_joins(district: :subscriptions)
      # Below is a slightly weird construction, but Rails magic doesn't work so we had to get explicit
      .where('(subscriptions.expiration >= ? OR subscriptions_districts.expiration >= ?)', DateTime.current, DateTime.current)
  end

  scope :premium, mylambda`

Reply:
I think you could rewrite with Rails methods only:

```
scope :premium, (lambda do
  left_outer_joins(:subscriptions)
    .left_outer_joins(district: :subscriptions)
    .where(
      subscriptions: { expiration: DateTime.current.. },
    )
    .or(
      left_outer_joins(:subscriptions)
        .left_outer_joins(district: :subscriptions)
        .where(districts: { subscriptions: { expiration: DateTime.current.. } })
    )
end)
```

Not sure if that's more readable though.